mlflow/models/utils.py

```diff
@@ -863,3 +866,38 @@ def get_model_version_from_model_uri(model_uri):
     (name, version) = get_model_name_and_version(client, model_uri)
     model_version = client.get_model_version(name, version)
     return model_version
+
+
+def _enforce_parameters_schema(params: Optional[Dict[str, Any]], schema: ParamSchema):
+    if params is None:
+        return params
+    if not isinstance(params, dict):
```

Comment:
Should we add an additional check here to ensure that the keys are all strings? It might be a bit confusing with the logger message in 881 if a key is invalidated if it should be a string-wrapped int but is actually an int due to the logging print parsing for this situation. (Error warning could be unclear as to why it's failing)